proxy/proxy.go

```diff
@@ -476,6 +479,21 @@ func (p *Proxy) Resolve(d *DNSContext) error {
 	return err
 }
 
+// Resolve and update the DNS cache.
+func (p *Proxy) resolveAndUpdateCache(d *DNSContext) {
+	defer func() {
+		if r := recover(); r != nil {
+			log.Error("Recovered in %s", r)
```

Comment:
I think a more informative message would be:

```
log.Error("resolveAndUpdateCache: recovered from panic: %v", r)
```

Reply:
fixed